src/libraries/System.Diagnostics.DiagnosticSource/ref/System.Diagnostics.DiagnosticSourceActivity.cs

```diff
@@ -121,4 +129,51 @@ public abstract partial class DiagnosticSource
         public System.Diagnostics.Activity StartActivity(System.Diagnostics.Activity activity, object? args) { throw null; }
         public void StopActivity(System.Diagnostics.Activity activity, object? args) { }
     }
+    public enum ActivityKind
+    {
+        Internal = 1,
+        Server = 2,
+        Client = 3,
+        Producer = 4,
+        Consumer = 5,
+    }
+    public readonly struct ActivityContext : IEquatable<ActivityContext>
+    {
+        public ActivityContext(System.Diagnostics.ActivityTraceId traceId, System.Diagnostics.ActivitySpanId spanId, System.Diagnostics.ActivityTraceFlags traceOptions, string? traceState = null) { throw null; }
+        public System.Diagnostics.ActivityTraceId TraceId { get; }
+        public System.Diagnostics.ActivitySpanId SpanId { get; }
+        public System.Diagnostics.ActivityTraceFlags TraceFlags { get; }
+        public string? TraceState { get; }
+        public static bool operator ==(System.Diagnostics.ActivityContext context1, System.Diagnostics.ActivityContext context2) { throw null; }
+        public static bool operator !=(System.Diagnostics.ActivityContext context1, System.Diagnostics.ActivityContext context2) { throw null; }
+        public bool Equals(System.Diagnostics.ActivityContext context) { throw null; }
+        public override bool Equals(object? obj) { throw null; }
+        public override int GetHashCode() { throw null; }
+    }
+    public readonly struct ActivityLink
+    {
+        public ActivityLink(System.Diagnostics.ActivityContext context) { throw null; }
+        public ActivityLink(System.Diagnostics.ActivityContext context, System.Collections.Generic.IDictionary<string, object>? attributes) { throw null; }
+        public System.Diagnostics.ActivityContext Context { get; }
+        public System.Collections.Generic.IDictionary<string, object>? Attributes { get; }
+    }
+    public sealed class ActivitySource : IDisposable
+    {
+        private ActivitySource() { throw null; }
+        public ActivitySource(string name) { throw null; }
+        public string Name { get; }
+        public Activity? StartActivity() { throw null; }
+        public Activity? StartActivity(System.Diagnostics.ActivityContext context, System.Collections.Generic.IEnumerable<ActivityLink>? links = null, System.DateTimeOffset startTime = default) { throw null; }
+        public void Dispose() { }
+        public static void AddListener(ActivityListener listener) {}
```

Comment:
This feels less discoverable to me? Ideally as an implementer of ActivityListener I don't need to know that the concept of ActivitySource exists. Maybe Subscribe() is an odd verb because it sounds like you are subscribing to the listener? It could also be ActivityListener.Enable(), ActivityListener.EnableCallbacks(), ActivityListener.Start()?

Reply:
I don't feel strongly about it but I have moved it to `ActivityListener.Start()`.